### PR TITLE
customcluster: support kubeversion to v1.26.5

### PR DESCRIPTION
**Pull request by Xieql**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

The previously supported kubeversion ranged from 'v1.22.0 - v1.24.6', and it has now been extended to include 'v1.22.0 - v1.26.5'.

part of #218 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/cluster-operator/customcluster_helper.go [hunk 1]
```diff
@@ -410,7 +410,12 @@ func (r *CustomClusterController) ensureWorkerPodCreated(ctx context.Context, cu
 		return workerPod, nil
 	}
 
-	newWorkerPod := generateClusterManageWorker(customCluster, manageAction, manageCMD, hostName, configName)
+	kubesprayImage, err := getKubesprayImage(kubeVersion)
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_helper.go:413]: dono we need to validate kubernetes version.
Then we will fall into dead loop when this function returns error

Comment (Xieql) [reply pkg/cluster-operator/customcluster_helper.go]: ok, I added  a log for it.

File: pkg/cluster-operator/customcluster_helper.go [hunk 2]
```diff
@@ -452,17 +450,22 @@ func (r *CustomClusterController) findManageWorkerPod(ctx context.Context, custo
 // Kubespray v2.20.0 supports Kubernetes versions from 1.22.0 to 1.24.6,
 // while Kubespray v2.22.1 supports Kubernetes versions from 1.24.0 to 1.26.5.
 // This function returns one of these two Kubespray versions based on the input Kubernetes version.
-func getKubesprayImage(kubeVersion string) (string, error) {
+func getKubesprayImage(ctx context.Context, kubeVersion string) string {
+	log := ctrl.LoggerFrom(ctx)
 	var kubesprayVersion string
 	kubeVersion = strings.TrimPrefix(kubeVersion, "v")
 	targetVersion, err := semver.NewVersion(kubeVersion)
+	// should not happen
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_helper.go:460]: why cannot it happen? Do we have validation on `kubeVersion`

Comment (Xieql) [reply pkg/cluster-operator/customcluster_helper.go:460]: you are right, we should validation on it

File: pkg/cluster-operator/customcluster_helper.go [hunk 3]
```diff
@@ -410,7 +411,9 @@ func (r *CustomClusterController) ensureWorkerPodCreated(ctx context.Context, cu
 		return workerPod, nil
 	}
 
-	newWorkerPod := generateClusterManageWorker(customCluster, manageAction, manageCMD, hostName, configName)
+	kubesprayImage := getKubesprayImage(ctx, kubeVersion)
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_helper.go:414]: What if this is empty

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

we have validation webhook now

**Pull request merged**